File: 001 Simple Hello World/Java/src/main/java/com/example/platestack/SimpleHelloWorld.java
package com.example.platestack;

import org.platestack.api.plugin.PlatePlugin;
import org.platestack.api.plugin.annotation.Plate;
import org.platestack.api.plugin.annotation.Version;

/**
 * The class must be public, must extends {@link PlatePlugin} and have a {@link Plate} annotation.
 *
 * It's recommended to specify every part of your version separately in your {@link Version} annotations
 * but it gets overcomplicated in pure java, so you can use {@link Version#value()} here.
 * <p>
 * Make sure to follow the <a href="http://semver.org/spec/v2.0.0.html">Semantic Versioning Specification 2.0.0</a>
 * or your plugin will fail to load.
 */
@Plate(id = "simple_java_plugin", name = "Simple Java Plugin", version = @Version("0.1.0-SNAPSHOT"))
public class SimpleHelloWorld extends PlatePlugin
{
    @Override
    protected void onEnable()
    {
        getLogger().info("Hello World from {} version {}", getMetadata().getName(), getVersion());
    }
}

File: 001 Simple Hello World/Kotlin/src/main/kotlin/SimpleHelloWorld.kt
package com.example.platestack // You don't need to have the same directory structure as your package
                               // But your IDE may yell at you with an ignorable warning

import org.platestack.api.plugin.PlatePlugin
import org.platestack.api.plugin.annotation.Plate
import org.platestack.api.plugin.annotation.Version

/**
 * Kotlin is more flexible then Java, no more then one object of your plugin may exists at the same time,
 * so you should use `object` instead of `class`. You can use `class` too only because Java doesn't
 * have this feature so Java plugins needs to use a normal `class` to work and since Kotlin is Java compatible
 * you can still do the Java-way unnecessarily. This may change in future.
 *
 * It's recommended to specify every part of your version separately in your [Version] annotations.
 * If you see hints like "major:", "minor:", "patch:", etc then you may want to disable it.
 * To disable just do these steps:
 * 1. Alt+Enter next to a hint
 * 2. click on "Do not show hints for current method"
 * 3. click on "Show Parameter Hints Settings"
 * 4. remove the parenthesis from Version to blacklist the entire annotation.
 */
@Plate("kotlin_obj_plugin", "Simple Kotlin Plugin", Version(0,1,0,"SNAPSHOT"), kotlin = "1.1.2-4")
object SimpleHelloWorld: PlatePlugin() {

    override fun onEnable() {
        logger.info { "Hello World from ${metadata.name} version $version" }
    }

}
